client/cache/random_expired_cache.go

```diff
@@ -43,10 +44,26 @@ func NewRandomExpireCache(adapter Cache, opts ...RandomExpireCacheOption) Cache
 	for _, fn := range opts {
 		fn(&rec)
 	}
+	if rec.Offset == nil {
```

Comment:
```go
	rec := RandomExpireCache {
            Cache: adapter,
            offset: defaultOffset()
        }
	rec.Cache = adapter
	for _, fn := range opts {
		fn(&rec)
	}
```

So you can remove this if block. 

I think using typical option pattern looks good